lib/screens/home_view_model/home_view_model.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:easel_flutter/easel_provider.dart';
+import 'package:easel_flutter/models/nft.dart';
+import 'package:easel_flutter/repository/repository.dart';
+import 'package:easel_flutter/utils/constants.dart';
+import 'package:easel_flutter/utils/enums.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/cupertino.dart';
+
+class HomeViewModel extends ChangeNotifier {
+  final Repository repository;
+
+  HomeViewModel(this.repository);
+
+  late ValueNotifier<int> currentPage;
+
+  late ValueNotifier<int> currentStep;
+  late PageController pageController;
+
+  NFT? nft;
+  String? from;
+  final List pageTitles = ["select_nft_file".tr(), "nft_detail_text".tr(), "nft_pricing".tr(), ''];
+
+  void init({required VoidCallback setTextField}) {
+    from = repository.getCacheString(key: fromKey);
+    repository.deleteCacheString(key: fromKey);
+
+    if (from == kDraft) {
+      nft = repository.getCacheDynamicType(key: nftKey);
+
+      Future.delayed(const Duration(milliseconds: 1), () {
+        setTextField.call();
+      });
+
+      if (nft!.step == UploadStep.assetUploaded.name) {
```

Comment:
this if/else if section should be refactored as it can be written more efficiently as well as use of constants.

Reply:
Done sir with switch.